bundles/org.openhab.binding.roborock/src/main/java/org/openhab/binding/roborock/internal/RoborockAccountHandler.java

```diff
@@ -0,0 +1,489 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link RoborockAccountHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+@NonNullByDefault
+public class RoborockAccountHandler extends BaseBridgeHandler implements MqttCallbackExtended {
+
+    private final Logger logger = LoggerFactory.getLogger(RoborockAccountHandler.class);
+
+    private final Storage<String> sessionStorage;
+    private @Nullable RoborockAccountConfiguration config;
+    private final SchedulerTask initTask;
+    private final SchedulerTask mqttConnectTask;
+    private final RoborockWebTargets webTargets;
+    private @Nullable MqttClient mqttClient;
+    private String token = "";
+    private String baseUri = "";
+    private Rriot rriot = new Login().new Rriot();
+    private final SecureRandom secureRandom = new SecureRandom();
+    private String mqttUser = "";
+    protected final Map<String, RoborockVacuumHandler> childDevices = new ConcurrentHashMap<>();
+    private final ExpiringCache<Home> homeCache = new ExpiringCache<>(Duration.ofMinutes(10), this::refreshHome);
+    private final ExpiringCache<HomeData> homeDataCache = new ExpiringCache<>(Duration.ofMinutes(10),
+            this::refreshHomeData);
+
+    private final Gson gson = new Gson();
+
+    public RoborockAccountHandler(Bridge bridge, Storage<String> stateStorage, HttpClient httpClient) {
+        super(bridge);
+        webTargets = new RoborockWebTargets(httpClient);
+        sessionStorage = stateStorage;
+        initTask = new SchedulerTask(scheduler, logger, "API Init", this::initAPI);
+        mqttConnectTask = new SchedulerTask(scheduler, logger, "MQTT Connection", this::establishMQTTConnection);
+    }
+
+    public String getToken() {
+        return token;
+    }
+
+    public ThingUID getUID() {
+        return thing.getUID();
+    }
+
+    @Nullable
+    public Home getHomeDetail() {
+        return homeCache.getValue();
+    }
+
+    @Nullable
+    public Home refreshHome() {
+        logger.info("refreshHome");
+        try {
+            logger.info("refreshHome2, baseuri = {}, token = {}", baseUri, token);
+            return webTargets.getHomeDetail(baseUri, token);
+        } catch (RoborockException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Error " + e.getMessage());
+            return null;
+        }
+    }
+
+    @Nullable
+    public HomeData getHomeData() {
+        return homeDataCache.getValue();
+    }
+
+    @Nullable
+    public HomeData refreshHomeData() {
+        logger.info("refreshHomeData");
+        try {
+            Home home = homeCache.getValue();
+            if (home == null) {
+                return new HomeData();
+            }
+            logger.info("refreshHomeData2");
```

Comment:
Please remove these kind of loggers, they provide no usefull information. They rather look like some debug/breakpoint replacement.
```suggestion
```

Reply:
caught this one earlier